cpp/oneapi/dal/backend/primitives/rng/rng.hpp

```diff
@@ -0,0 +1,152 @@
+/*******************************************************************************
+* Copyright contributors to the oneDAL project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/backend/primitives/rng/engine_cpu.hpp"
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+#include "oneapi/dal/backend/primitives/rng/engine_gpu.hpp"
+
+#endif
+
+namespace oneapi::dal::backend::primitives {
+template <typename Type, typename Size = std::int64_t>
+class rng {
```

Comment:
What is the purpose of this class? Why all the public methods cannot be just free non-member functions?
Previously it was holding daal engine as a state, but now there is no state and no purpose in grouping the methods together.